### ISSUE TITLE
Incompatible shapes during training

### ISSUE THREAD

**Issue by aklam** [lvapeab/nmt-keras#91]
Hi!
I get the following error: InvalidArgumentError (see above for traceback): Incompatible shapes: [20,30,30000] vs. [20,29,30000]
	 [[node loss/target_text_loss/mul (defined at /home/ec2-user/nmt-keras/src/keras/keras/backend/tensorflow_backend.py:3764)  = Mul[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"](_arg_target_text_target_0_3/_377, loss/target_text_loss/Log)]]

I expect this has to do with the Dataset i created and variable length sentences. I left all the training parameters as default, except for:
params['INPUT_VOCABULARY_SIZE'] = ds.vocabulary_len['source_text']
params['OUTPUT_VOCABULARY_SIZE'] = ds.vocabulary_len['target_text']

Below I define my dataset, note Cornell_train_reply.en is a file that contains sentences on each line.

`import sys
sys.path.append('../nmt-keras')
sys.path.append('../nmt-keras/nmt_keras')

from data_engine.prepare_data import keep_n_captions
from keras_wrapper.dataset import Dataset, saveDataset

dataset_name = 'Cornell_base'
cornell_path = '/data'

ds = Dataset(dataset_name, cornell_path, silence=True)

ds.setInput('data/Cornell_train_query.en',
    'train',
    type='text',
    id='source_text',
    pad_on_batch=True,
    tokenization='tokenize_basic',
    build_vocabulary=True,
    fill='end',
    max_text_len=30,
    max_words=30000,
    min_occ=0)

ds.setInput('data/Cornell_valid_query.en',
    'val',
    type='text',
    id='source_text',
    pad_on_batch=True,
    tokenization='tokenize_basic',
    fill='end',
    max_text_len=30,
    max_words=0)

ds.setInput('data/Cornell_train_reply_offset.en',
    'train',
    type='text',
    id='state_below',
    required=False,
    tokenization='tokenize_basic',
    pad_on_batch=True,
    build_vocabulary='target_text',
    offset=1,
    fill='end',
    max_text_len=30,
    max_words=30000)

ds.setInput(None, 
    'val',
    type='ghost',
    id='state_below',
    pad_on_batch=True,
    required=False)

ds.setOutput('data/Cornell_train_reply.en',
    'train',
    type='text',
    id='target_text',
    tokenization='tokenize_basic',
    build_vocabulary=True,
    pad_on_batch=True,
    sample_weights=True,
    max_text_len=30,
    max_words=30000,
    min_occ=0)

ds.setOutput('data/Cornell_valid_reply.en',
    'val',
    type='text',
    id='target_text',
    pad_on_batch=True,
    tokenization='tokenize_basic',
    sample_weights=True,
    max_text_len=30,
    max_words=0)

#ds.merge_vocabularies(['source_text', 'target_text'])

keep_n_captions(ds, repeat=1, n=1, set_names=['val'])

saveDataset(ds, 'query_to_reply')

`

**Comment by lvapeab** [comment]
Hi!

It looks like a regression. Thanks for reporting it. I've just updated the keras_wrapper (https://github.com/lvapeab/multimodal_keras_wrapper/commit/e02b46853b1d6728a997db4ec138698ec6bfef04) and it works. 

Please, update this repo (`cd /home/ec2-user/nmt-keras/src/keras-wrapper; git pull`) and try again.

Cheers!
